### PR TITLE
[TwigBridge] use proper class to fetch asset version strategy property

### PR DESCRIPTION
**Pull request by xabbuh**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #14692 |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

**Comment by iltar** [comment]
I'll see if this fixes the issue tomorrow, thanks for the quick patch!

Lines 110 and 113 feature `$v->setAccessible(true);`, this could be pushed down to just below the if-statement.

**Comment by xabbuh** [comment]
@iltar good catch, thanks

**Comment by dosten** [comment]
:+1:

File: src/Symfony/Bridge/Twig/Extension/AssetExtension.php [hunk 1]
```diff
@@ -99,9 +99,19 @@ private function getLegacyAssetUrl($path, $packageName = null, $absolute = false
...
+                $class = new \ReflectionClass($package);
+
+                do {
+                    $class = $class->getParentClass();
+                 } while ('Symfony\Component\Asset\Package' !== $class->getName());
 
+                $v = $class->getProperty('versionStrategy');
+            } else {
+                $v = new \ReflectionProperty($package, 'versionStrategy');
+            }
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/Twig/Extension/AssetExtension.php]: You could simplify the whole if/else block with:
```php
$class = new \ReflectionClass($package);

while ('Symfony\Component\Asset\Package' !== $class->getName()) {
    $class = $class->getParentClass();
}

v = $class->getProperty('versionStrategy'); 
```

**Comment by nicolas-grekas** [comment]
:+1:

File: src/Symfony/Bridge/Twig/Extension/AssetExtension.php (hunk 1 above)
Comment (xabbuh) [inline src/Symfony/Bridge/Twig/Extension/AssetExtension.php]: indeed, done

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**